spark-3.0/src/main/scala/org/neo4j/spark/writer/Neo4jDataWriter.scala

```diff
@@ -0,0 +1,130 @@
+package org.neo4j.spark.writer
+import java.util
+import java.util.concurrent.CountDownLatch
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.SaveMode
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.connector.write.{DataWriter, WriterCommitMessage}
+import org.apache.spark.sql.types.StructType
+import org.neo4j.driver.exceptions.{ClientException, Neo4jException, ServiceUnavailableException, SessionExpiredException}
+import org.neo4j.driver.{Session, Transaction, Values}
+import org.neo4j.spark.service.{MappingService, Neo4jQueryService, Neo4jQueryStrategy, Neo4jQueryWriteStrategy, Neo4jWriteMappingStrategy}
+import org.neo4j.spark.util.{DriverCache, Neo4jOptions, Neo4jUtil}
+import org.neo4j.spark.util.Neo4jUtil.closeSafety
+
+import scala.collection.JavaConverters._
+
+class Neo4jDataWriter(jobId: String,
```

Comment:
the code seems very similar to 2.4 version maybe makes sens having a common class extended by the data source writer in the related modules, wdyt?